UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs

```diff
@@ -474,8 +474,13 @@ public void CmdRespawnPlayer()
 	}
 
 	[Server]
-	public void ServerRespawnPlayer()
+	public void ServerRespawnPlayer(string occupation = null)
 	{
+		if (occupation != null)
+		{
+			playerScript.mind.occupation = Resources.Load<Occupation>($"ScriptableObjects/Occupations/"+occupation);
```

Comment:
could you use the SingletonScriptableObject to get the references from, you will have to add it to the SingletonScriptableObject  loader script

Reply:
ScriptableObjectSingletons in there

Reply:
Yes, I can do that. Would it work if I add a scriptable object that contains a list of occupations? So we don’t have to add all jobs to the Singleton list

Reply:
done